### DISCUSSION TITLE
Drone crash when changing the frequency of sending setpoints from CFLib in /examples/autonomy/autonomousSequence.py

### DISCUSSION THREAD

**Discussion by nico-kng** [bitcraze/discussions#1376] [Q&A] [answered]
Hi, I am currently working on my master thesis and I am trying to understand the bitcraze firmware. I am currently looking at the options with which I can send commands from the CFLib to the drone. For this I use an example (/examples/autonomy/autonomousSequence.py) from the CFLib. In the example, a new position is sent to the drone every 5 seconds. The function that sends the setpoint to the drone is executed with a frequency of 10 Hz. 

https://github.com/bitcraze/crazyflie-lib-python/blob/c7eaabe31612e796fcdcf8511c18d197051e17cd/examples/autonomy/autonomousSequence.py#L127-L137

If I reduce the frequency in line 137 (from 10 Hz to 5 Hz) by setting the time the program goes to sleep to 200 ms, the drone crashes. Unfortunately, I have not found any documentation for the transmission interval. Is there a technical reason for this crash in the firmware?

**Comment by ToveRumar** [comment]
Hi!

I don't see any immediate problem with this, so i just need to know a bit more about the issue. 

Do you mean that you see a Firmware crash (If so, could you please provide a log or line number for the crash), or a physical crash? 

Have you set it up with the Loco positioning system? Are you using any additional decks?

**Reply by nico-kng** [reply]
Hi, I'm talking about a physical crash. I use the lighthouse system for positioning. As it turned out, the crash happened because of a drone error. When I retested, the change in frequency worked. Analyzing the code further, I came across a safety function. In the supervisor module there is a query about the age of the last setpoint. 

https://github.com/bitcraze/crazyflie-firmware/blob/b15eebc864e7e734daa3766d116fa26edbc940a9/src/modules/src/supervisor.c#L318C1-L323C4

This query seems to be the reason why you have to send the setpoint repeatedly with the CFLib. A possible scenario would be if the drone was flying outside the transmission range of the radio. Then it would no longer be possible to stop the drone. The drone locks itself by the query if no new packet is sent from the host computer with a setpoint in 2000 ticks (= 2 seconds). Are there other reasons for this behavior?

**Reply by ToveRumar** [reply, marked as answer]
Hi!

Yes, the drone will level out if it stops receiving packets from the host.
The example you are referring to uses the low level commander which needs to receive set points with a certain frequency.
This is for safety reasons, so that you will not loose the drone if communication is dropped for some reason. After packet loss it should first level out and then stop the motors after a while. 

If you do not want to update the set point from the host with this frequency you need to use a high level commander, on board the drone. You can see how this can be done in this example; https://github.com/bitcraze/crazyflie-lib-python/blob/c7eaabe31612e796fcdcf8511c18d197051e17cd/examples/autonomy/autonomous_sequence_high_level.py

The high level commander on board will then feed the low level commander with set points at a high frequency, so that it will not level out even though you are not updating the set point from the host as often. 

Did this answer your question?
